### PR TITLE
Fix for #988 Codeception report failed when Test format is used

### PR DESCRIPTION
**Pull request by tankist**
Added class::method fallback for feature entry to prevent empty report for Codeception plugin

### CODE REVIEW

File: PHPCI/Plugin/Util/TestResultParsers/Codeception.php [hunk 1]
```diff
@@ -62,6 +62,11 @@ public function parse()
                     $testresult['class'] = (string) $testcase['class'];
                 }
 
+                //PHPUnit testcases does not have feature field. Use class::method instead
```
Comment (mavimo) [inline PHPCI/Plugin/Util/TestResultParsers/Codeception.php]: Space between ```//``` and comment.

File: PHPCI/Plugin/Util/TestResultParsers/Codeception.php [hunk 2]
```diff
@@ -50,10 +50,10 @@ public function parse()
 
             foreach ($testsuite->testcase as $testcase) {
                 $testresult = array(
-                    'suite' => (string) $testsuite['name'],
+                    'suite' => ucfirst((string) $testsuite['name']),
                     'file' => str_replace($this->phpci->buildPath, '/', (string) $testcase['file']),
                     'name' => (string) $testcase['name'],
-                    'feature' => (string) $testcase['feature'],
+                    'feature' => ucfirst((string) $testcase['feature']),
```
Comment (mavimo) [inline PHPCI/Plugin/Util/TestResultParsers/Codeception.php]: Why UCFirst?

File: PHPCI/Plugin/Util/TestResultParsers/Codeception.php [hunk 3]
```diff
@@ -50,10 +50,10 @@ public function parse()
 
             foreach ($testsuite->testcase as $testcase) {
                 $testresult = array(
-                    'suite' => (string) $testsuite['name'],
+                    'suite' => ucfirst((string) $testsuite['name']),
```
Comment (mavimo) [inline PHPCI/Plugin/Util/TestResultParsers/Codeception.php]: Why UCFirst?

File: PHPCI/Plugin/Util/TestResultParsers/Codeception.php (hunk 2 above)
Comment (tankist) [inline PHPCI/Plugin/Util/TestResultParsers/Codeception.php]: Just for pretty output. "feature" attribute is taken by Codeception from the method name and in most cases this method name is started from lower case letter.

**Comment by tankist** [comment]
Writing my last comment I have noticed that "feature" attribute is taken from test method name. For my quickfix if "feature" attribute is empty I'm filling it as class::method. So what do you think, @mavimo, maybe it's better to add logic to split method name as well as Codeception do it for Cest classes?

I mean I do in this fix: "Unit\Library\Ras\Logger\Adapter\LimitedFileTest::testCanCreateLimitedFileAdapter"
What can I do: "Test can create limited file adapter"

In second case we will have unified Codeception log for both Cest/Test formats.

File: PHPCI/Plugin/Util/TestResultParsers/Codeception.php (hunk 1 above)
Comment (tankist) [inline PHPCI/Plugin/Util/TestResultParsers/Codeception.php]: Updated

**Comment by mavimo** [comment]
@tankist since you are introducing two different changes (formatting cleanup and failure when Test format is used) i prefer to split this MR in two part. The first one only fix the Test format report (and I think is fine) for the formatting I'm not sure is "good" alter the codexception result since can be "source of issues".

**Comment by tankist** [comment]
So you mean remove "ucfirst" and leave if block only?

**Pull request closed without merging**